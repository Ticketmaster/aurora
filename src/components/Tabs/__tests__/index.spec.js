import React from "react";
import { render, Simulate } from "react-testing-library";

import { Tabs } from "../index";

const items = ["Test tab item 1", "Test tab item 2", "Test tab item 3"];
const onTabsClick = jest.fn();
const mainProps = {
  items,
  index: 0,
  onClick: onTabsClick
};
const textProps = {
  variant: "accent",
  accent: "alert",
  size: "giga",
  weight: "semiBold"
};

describe("<Tabs />", () => {
  it("should render tabs component correctly", () => {
    const { container, unmount } = render(<Tabs {...mainProps} />);
    expect(container).toMatchSnapshot();
    unmount();
  });

  it("should render tabs component correctly when withBorderBottom equals false", () => {
    const { container, unmount } = render(
      <Tabs {...mainProps} withBorderBottom={false} />
    );
    expect(container).toMatchSnapshot();
    unmount();
  });

  it("should render tabs component correctly with props for text of tabs", () => {
    const { container, unmount } = render(
      <Tabs {...mainProps} {...textProps} />
    );
    expect(container).toMatchSnapshot();
    unmount();
  });

  it("should call onClick function with index of tab when click on it", () => {
    const { unmount, container } = render(<Tabs {...mainProps} />);
    Simulate.click(container.querySelector('div[data-index="1"]'));
    expect(onTabsClick).toHaveBeenLastCalledWith(1);
    unmount();
  });
});
