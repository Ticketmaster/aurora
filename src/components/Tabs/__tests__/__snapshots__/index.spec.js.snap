// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> should render tabs component correctly 1`] = `
.c4 {
  font-size: 16px;
  line-height: 1.5;
  text-transform: none;
  color: rgba(38,38,38,1);
}

.c0 {
  position: relative;
  z-index: 1;
  width: 100%;
  overflow: visible;
}

.c0:after {
  content: '';
  display: block;
  position: absolute;
  border-bottom: 1px solid #ebebeb;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: -1;
}

.c0.tabs__container--overflowed:after {
  width: 100vw;
  left: calc(50% - 50vw);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  width: 100%;
  overflow-x: auto;
  -webkit-scroll-behavior: smooth;
  -moz-scroll-behavior: smooth;
  -ms-scroll-behavior: smooth;
  scroll-behavior: smooth;
}

.c1::-webkit-scrollbar {
  display: none;
}

.c1.tab__content--overflowed {
  margin-left: calc(50% - 50vw);
  padding-left: calc(50vw - 50%);
  width: 100vw;
}

.c2 {
  display: block;
  padding-right: 24px;
  white-space: nowrap;
}

.c2:last-of-type {
  padding-right: 0;
}

.c3 {
  padding: 16px 0 8px 0;
  cursor: pointer;
  background-color: transparent;
  border: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c3.tab__button--active {
  border-bottom: 4px solid #026cdf;
}

@media screen and (min-width:480px) {
  .c4 {
    font-size: 16px;
  }
}

@media screen and (min-width:1024px) {
  .c4 {
    font-size: 20px;
  }
}

@media screen and (min-width:480px) {
  .c0.tabs__container--overflowed:after {
    width: 100%;
    left: 0;
  }
}

@media screen and (min-width:480px) {
  .c1.tab__content--overflowed {
    margin-left: 0;
    padding-left: 0;
    width: 100%;
  }
}

@media screen and (min-width:1024px) {
  .c2 {
    padding-right: 32px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      role="tablist"
    >
      <div
        class="c2"
      >
        <button
          aria-selected="true"
          class="tab__button--active c3"
          data-index="0"
          role="tab"
        >
          <div
            class="text text--primary c4"
          >
            Test tab item 1
          </div>
        </button>
      </div>
      <div
        class="c2"
      >
        <button
          aria-selected="false"
          class="c3"
          data-index="1"
          role="tab"
        >
          <div
            class="text text--primary c4"
          >
            Test tab item 2
          </div>
        </button>
      </div>
      <div
        class="c2"
      >
        <button
          aria-selected="false"
          class="c3"
          data-index="2"
          role="tab"
        >
          <div
            class="text text--primary c4"
          >
            Test tab item 3
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Tabs /> should render tabs component correctly when withBorderBottom equals false 1`] = `
.c4 {
  font-size: 16px;
  line-height: 1.5;
  text-transform: none;
  color: rgba(38,38,38,1);
}

.c0 {
  position: relative;
  z-index: 1;
  width: 100%;
  overflow: visible;
}

.c0.tabs__container--overflowed:after {
  width: 100vw;
  left: calc(50% - 50vw);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  width: 100%;
  overflow-x: auto;
  -webkit-scroll-behavior: smooth;
  -moz-scroll-behavior: smooth;
  -ms-scroll-behavior: smooth;
  scroll-behavior: smooth;
}

.c1::-webkit-scrollbar {
  display: none;
}

.c1.tab__content--overflowed {
  margin-left: calc(50% - 50vw);
  padding-left: calc(50vw - 50%);
  width: 100vw;
}

.c2 {
  display: block;
  padding-right: 24px;
  white-space: nowrap;
}

.c2:last-of-type {
  padding-right: 0;
}

.c3 {
  padding: 16px 0 8px 0;
  cursor: pointer;
  background-color: transparent;
  border: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c3.tab__button--active {
  border-bottom: 4px solid #026cdf;
}

@media screen and (min-width:480px) {
  .c4 {
    font-size: 16px;
  }
}

@media screen and (min-width:1024px) {
  .c4 {
    font-size: 20px;
  }
}

@media screen and (min-width:480px) {
  .c0.tabs__container--overflowed:after {
    width: 100%;
    left: 0;
  }
}

@media screen and (min-width:480px) {
  .c1.tab__content--overflowed {
    margin-left: 0;
    padding-left: 0;
    width: 100%;
  }
}

@media screen and (min-width:1024px) {
  .c2 {
    padding-right: 32px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      role="tablist"
    >
      <div
        class="c2"
      >
        <button
          aria-selected="true"
          class="tab__button--active c3"
          data-index="0"
          role="tab"
        >
          <div
            class="text text--primary c4"
          >
            Test tab item 1
          </div>
        </button>
      </div>
      <div
        class="c2"
      >
        <button
          aria-selected="false"
          class="c3"
          data-index="1"
          role="tab"
        >
          <div
            class="text text--primary c4"
          >
            Test tab item 2
          </div>
        </button>
      </div>
      <div
        class="c2"
      >
        <button
          aria-selected="false"
          class="c3"
          data-index="2"
          role="tab"
        >
          <div
            class="text text--primary c4"
          >
            Test tab item 3
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Tabs /> should render tabs component correctly with props for text of tabs 1`] = `
.c4 {
  font-size: 16px;
  font-weight: 600;
  line-height: 1.5;
  text-transform: none;
  color: #AE393A;
}

.c0 {
  position: relative;
  z-index: 1;
  width: 100%;
  overflow: visible;
}

.c0:after {
  content: '';
  display: block;
  position: absolute;
  border-bottom: 1px solid #ebebeb;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: -1;
}

.c0.tabs__container--overflowed:after {
  width: 100vw;
  left: calc(50% - 50vw);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  width: 100%;
  overflow-x: auto;
  -webkit-scroll-behavior: smooth;
  -moz-scroll-behavior: smooth;
  -ms-scroll-behavior: smooth;
  scroll-behavior: smooth;
}

.c1::-webkit-scrollbar {
  display: none;
}

.c1.tab__content--overflowed {
  margin-left: calc(50% - 50vw);
  padding-left: calc(50vw - 50%);
  width: 100vw;
}

.c2 {
  display: block;
  padding-right: 24px;
  white-space: nowrap;
}

.c2:last-of-type {
  padding-right: 0;
}

.c3 {
  padding: 16px 0 8px 0;
  cursor: pointer;
  background-color: transparent;
  border: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c3.tab__button--active {
  border-bottom: 4px solid #026cdf;
}

@media screen and (min-width:480px) {
  .c4 {
    font-size: 16px;
  }
}

@media screen and (min-width:1024px) {
  .c4 {
    font-size: 20px;
  }
}

@media screen and (min-width:480px) {
  .c0.tabs__container--overflowed:after {
    width: 100%;
    left: 0;
  }
}

@media screen and (min-width:480px) {
  .c1.tab__content--overflowed {
    margin-left: 0;
    padding-left: 0;
    width: 100%;
  }
}

@media screen and (min-width:1024px) {
  .c2 {
    padding-right: 32px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      role="tablist"
    >
      <div
        class="c2"
      >
        <button
          aria-selected="true"
          class="tab__button--active c3"
          data-index="0"
          role="tab"
        >
          <div
            class="text text--accent text--alert text--primary c4"
          >
            Test tab item 1
          </div>
        </button>
      </div>
      <div
        class="c2"
      >
        <button
          aria-selected="false"
          class="c3"
          data-index="1"
          role="tab"
        >
          <div
            class="text text--accent text--alert text--primary c4"
          >
            Test tab item 2
          </div>
        </button>
      </div>
      <div
        class="c2"
      >
        <button
          aria-selected="false"
          class="c3"
          data-index="2"
          role="tab"
        >
          <div
            class="text text--accent text--alert text--primary c4"
          >
            Test tab item 3
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
